### PR TITLE
6/7/2015 global timer, fixes issue #11

### PR DESCRIPTION
**Pull request by princesspretzel**
Added timer button, related methods, and model attribute.

Helpful screenshot:
screen shot 2015-06-07 at 9 01 40 pm

### CODE REVIEW

File: public/css/meeting.css [hunk 1]
```diff
@@ -4,6 +4,34 @@ html, body {
   color: #212121;
 }
 
+.brittanybutton {
```
Comment (typpo) [inline public/css/meeting.css]: Can you give it a more descriptive name?

Comment (typpo) [inline public/css/meeting.css]: Also maybe you should just use a bootstrap button.   "btn btn-primary" or similar.

File: public/css/meeting.css [hunk 2]
```diff
@@ -4,6 +4,34 @@ html, body {
...
+
+.brittanybutton:hover {
+  background-color:#5cbf2a;
+}
+.brittanybutton:active {
+  position:relative;
+  top:1px;
+}
+
+#startbutton {
```
Comment (typpo) [inline public/css/meeting.css]: remove?

File: public/js/main.js [hunk 3]
```diff
@@ -26,6 +30,14 @@ function App() {
     return socket;
   };
 
+  App.prototype._initTime = function() {
+    var socket = io();
+    socket.emit('get total time', this.getChatId());
```
Comment (typpo) [inline public/js/main.js]: Doesn't need to be in this change, but startTimer is going to have to save a "timeStarted" field in the Meeting, and getting the time from server will have to subtract time elapsed from total time.  In order to support the case where someone joins mid-meeting.

File: public/js/timer.js [hunk 4]
```diff
@@ -1,45 +1,40 @@
...
+    minutes = minutes < 10 ? "0" + minutes : minutes;
+    seconds = seconds < 10 ? "0" + seconds : seconds;
+
+    display.textContent = minutes + ":" + seconds;
+
+    if (diff <= 0) {
+      // add one second so that the count down starts at the full duration
+      // example 05:00 not 04:59
+      //start = Date.now() + 1000;
+      t0Callback();
```
Comment (typpo) [inline public/js/timer.js]: Should create a bug that time goes negative if it runs over I think.

File: routes.js [hunk 5]
```diff
@@ -78,6 +79,32 @@ module.exports = function(app,io) {
       // TODO associate meeting with socket.
     });
 
+    /** Timer stuff below **/
+
+    socket.on('get total time', function (id) {
+      var mtg = new Meeting()
+      var query = mtg.findOne({id: id})
```
Comment (typpo) [inline routes.js]: Meeting.findOne

no need to create dummy mtg

File: routes.js [hunk 6]
```diff
@@ -78,6 +79,32 @@ module.exports = function(app,io) {
...
+
+      query.exec(function (err, querymtg) {
+        if (err) {
+          socket.emit('error', id);
+          return;
+        }
+        socket.emit('get total time', querymtg.meetingLength);
+        socket.broadcast.to(this.room).emit('get total time', querymtg.meetingLength);
```
Comment (typpo) [inline routes.js]: Does this definitely work?  "this" keyword inside the closure is different from this = socket, which is how it's used elsewhere.

File: public/css/meeting.css (hunk 1 above)
Comment (typpo) [inline public/css/meeting.css]: On second thought your button is pretty awesome, maybe we should keep something like it :)

Comment (princesspretzel) [inline public/css/meeting.css]: WE KEEPS IT

File: public/js/main.js (hunk 3 above)
Comment (princesspretzel) [inline public/js/main.js]: That makes sense, but I think we should wait on it until we are supporting all mid-meeting joining-related activities.

File: public/js/timer.js (hunk 4 above)
Comment (princesspretzel) [inline public/js/timer.js]: A bug to disallow that?

File: routes.js (hunk 6 above)
Comment (princesspretzel) [inline routes.js]: No, I have to wait until the /createmeeting endpoint works and we have a db to test it.

Comment (typpo) [inline routes.js]: Oh ok.  I can tell you that there's a bug with your closure.  You should set

var sock = this;

after the var query line, and then .to(sock.room)

File: public/js/main.js [hunk 7]
```diff
@@ -2,10 +2,14 @@ function App() {
   var me = this;
   App.prototype.init = function() {
     var socket = me._initSocket();
+    var totalTime = me._initTime();
     agenda.init(this, socket, $('.agenda'));
     chat.init(this, socket, $('.chat'));
     video.init(this, $('#left-col'));
     uploader.init(this, socket, $('#file_input'));
+    document.querySelector('.timer-button').onclick = function(){
```
Comment (typpo) [inline public/js/main.js]: fyi jquery shorthand is $('.timer-button').on('click', function() {

**Pull request merged**